### ISSUE TITLE
I have some questions.

### ISSUE THREAD

**Issue by Cwwwwwx** [YCWu-nk/Cell_SAM#1]
Dear author,

I would like to express my gratitude for your valuable work. I would like to know 1. how to visualize the results of fine-tune SAM, there doesn't seem to be well-running test code in https://github.com/tianrun-chen/SAM-Adapter-PyTorch, and 2. when using the prompt SAM, I found that my graphics card displayed insufficient memory when there were more targets, but it ran well when there were few targets, and what suggestions do you have for this? I am interested in applying this code to my own experiments and would greatly appreciate your response.

Thank you very much for your assistance in advance!

**Comment by YCWu-nk** [comment]
Thank you for your interest in our work.

1.You can use this code repository directly, as we have changed a few things to make the finetune process more stable, and use save.py instead of test.py, which provides the function of saving results, as described in instructions.
2.There are two methods. You can process each small block of the image in sequence, or use the vit_b instead of vit_h version of SAM. Regarding the decline in performance in your data, you need to evaluate it yourself.

**Comment by Cwwwwwx** [comment]
Thank you, your suggestions are work. Also, a small additional question that prompt SAM saves the results with white padding, what problems will occur if these white are cancelled？

**Comment by YCWu-nk** [comment]
Usually, these paddings are not necessary. Replace draw_and_save with the following code：
```
from PIL import Image, ImageDraw 
def draw_and_save(image, pred_dict, save_path, random_color=True, show_label=False):
    output_image = Image.fromarray(image)
    with_mask = 'masks' in pred_dict
    labels = pred_dict['labels']
    scores = pred_dict['scores']
    bboxes = pred_dict['boxes'].cpu().numpy()

    draw = ImageDraw.Draw(output_image) 
    for box, label, score in zip(bboxes, labels, scores):
        x0, y0 = int(box[0]), int(box[1])
        w, h = int(box[2] - box[0]), int(box[3] - box[1])
        if show_label:
            draw.text((x0, y0 - 10), f'{label}', fill=(255, 255, 255))

    if with_mask:
        masks = pred_dict['masks'].cpu().numpy()
        for i, mask in enumerate(masks):
            mask = np.squeeze(mask)
            if mask.ndim == 2:  
                mask_image = Image.fromarray((mask * 255).astype(np.uint8)).convert('L')
                mask_image = mask_image.point(lambda p: p > 128 and 255) 
                if pred_dict['labels'][i] == '1':
                    color = (30, 144, 255)
                else:
                    color = (255, 0, 138)
                output_image.paste(Image.new('RGB', output_image.size, color), mask=mask_image)

    output_image.save(save_path)
```
Or use the batch cutting script：
```
from PIL import Image
import os
import unicodedata

def normalize_filename(filename):
    return unicodedata.normalize('NFKD', filename).encode('ascii', 'ignore').decode('ascii')

def crop_and_scale_images(input_folder, output_folder):
    if not os.path.exists(output_folder):
        os.makedirs(output_folder)

    for filename in os.listdir(input_folder):
        if filename.lower().endswith(('.png', '.jpg', '.jpeg')):
            image_path = os.path.join(input_folder, filename)
            img = Image.open(image_path)

            left = 128
            top = 120
            right = 898
            bottom = 890

            cropped_img = img.crop((left, top, right, bottom))

            scaled_img = cropped_img.resize((1000, 1000), Image.LANCZOS)

            normalized_filename = normalize_filename(filename)
            output_path = os.path.join(output_folder, normalized_filename)
            scaled_img.save(output_path)

input_folder = r'xxx' 
output_folder = r'xxx' 
crop_and_scale_images(input_folder, output_folder)
```
